File: CoreTransformer/TransitionSpatialMultiplierCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports System.Collections.ObjectModel

Friend Class TransitionSpatialMultiplierCollection
    Inherits KeyedCollection(Of Integer, TransitionSpatialMultiplier)

    Protected Overrides Function GetKeyForItem(ByVal item As TransitionSpatialMultiplier) As Integer
        Return item.TransitionSpatialMultiplierId
    End Function

End Class

File: src/Runtime/TransitionSpatialMultiplierCollection.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class TransitionSpatialMultiplierCollection : KeyedCollection<int, TransitionSpatialMultiplier>
    {
        protected override int GetKeyForItem(TransitionSpatialMultiplier item)
        {
            return item.TransitionSpatialMultiplierId;
        }
    }
}
